purchase_order_supplierinfo_update/models/purchase_order.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def write(self, vals):
```

Comment:
Why don't extend button_approve instead of write:
https://github.com/odoo/odoo/blob/d9ff2e8ec7200aa15830e92e5083251c4e81dd1c/addons/purchase/models/purchase.py#L349

What happen when the user clicks in Lock button in an approved PO?
I think that unnecesary update_supplierinfo_prince will be called.

Reply:
@carlosdauden check the changes I made regarding this, please.